File: src/eu/yeger/gramofo/fol/formula/FOLConstant.java
package eu.yeger.gramofo.fol.formula;

public class FOLConstant extends FOLFormula {

    FOLConstant(String name) {
        super(FOLType.Constant, null, false, false, name);
    }

    @Override
    public String getFormulaStringForDepth(int currentDepth, int maxDepth) {
        if (currentDepth >= maxDepth && maxDepth != -1) {
            return DOT;
        }
        return getName();
    }

    @Override
    public String toString() {
        return getName();
    }
}

File: src/eu/yeger/gramofo/fol/formula/FOLConstant.kt
package eu.yeger.gramofo.fol.formula

import eu.yeger.gramofo.model.domain.Node

sealed class FOLConstant(name: String) : FOLFormula(
    type = FOLType.Constant,
    name = name
) {
    class True : FOLConstant(TT)
    class False : FOLConstant(FF)

    override fun getFormulaString(variableAssignments: Map<String, Node>): String {
        return name
    }
}
